compiler/resolution/functionResolution.cpp

```diff
@@ -7110,7 +7110,7 @@ void resolveInitVar(CallExpr* call) {
              targetType->getValType()->symbol->hasFlag(FLAG_ARRAY) ||
              isDomainWithoutNew ||
              initCopySyncSingle ||
-             srcType->getValType()->symbol->hasFlag(FLAG_TUPLE) ||
+             (targetType->getValType()->symbol->hasFlag(FLAG_TUPLE) && srcType->getValType()->symbol->hasFlag(FLAG_TUPLE)) ||
```

Comment:
This looks too long for an 80 character line - I think you can split it on the `&&`

Reply:
:+1: